library/Network/Wai/Cli.hs

```diff
@@ -88,13 +92,29 @@ instance Options WaiOptions where
 
 
 #ifdef WaiCliUnix
+
+fdStart :: CInt
+fdStart = 3
+
+getActivatedSockets :: IO (Maybe [S.Socket])
+getActivatedSockets = runMaybeT $ do
+    listenPid <- read <$> MaybeT (getEnv "LISTEN_PID")
+    listenFDs <- read <$> MaybeT (getEnv "LISTEN_FDS")
+    myPid     <- lift getProcessID
+    guard $ listenPid == myPid
+    mapM makeSocket [fdStart .. fdStart + listenFDs - 1]
+  where
```

Comment:
We really don't need a `where` block here. `mapM (lift . S.mkSocket) […]` or something?